cryptoservice/crypto_service.go

```diff
@@ -110,28 +100,23 @@ func (ccs *CryptoService) Sign(keyIDs []string, payload []byte) ([]data.Signatur
 			}
 		}
 
-		algorithm := privKey.Algorithm()
 		var sigAlgorithm data.SigAlgorithm
-		var sig []byte
 
-		switch algorithm {
-		case data.RSAKey:
-			sig, err = rsaSign(privKey, payload)
+		switch privKey.(type) {
+		case *data.RSAPrivateKey, *data.RSAx509PrivateKey:
```

Comment:
I don't think we currently use it anywhere - or, at least, the tests seem to pass if they're removed?